tex/polyglossia-punct.lua

```diff
@@ -150,7 +208,21 @@ local function process(head)
                                 head = remove_node(head, next)
                             end
                         end
-                        head, start = insert_node_after(head, start, get_kern_node(rightspace*quad))
+                        if rightspace.unit == "quad" then
+                            unit = fontparameters.quad
+                            spacing_node = get_kern_node(rightspace.kern*unit)
+                        elseif rightspace.unit == "space" then
+                            unit = fontparameters.space
+                            if rightspace.rubber then
+                                stretch = rightspace.kern*fontparameters.space_stretch
+                                shrink  = rightspace.kern*fontparameters.space_shrink
+                                spacing_node = get_glue_node(rightspace.kern*unit, stretch, shrink)
+                                head = insert_node_after(head, start, get_penalty_node())
```

Comment:
```lua
head, start = insert_node_after(head, start, ...)
```
Like the code shown, `start` seems to be needed. Otherwise, we will get `glue, penalty` sequence instead of `penalty, glue`.